crates/database/src/order_execution.rs

```diff
@@ -18,25 +18,26 @@ pub async fn save(
     order: &OrderUid,
     auction: AuctionId,
     block_number: i64,
-    executed_fee: &BigDecimal,
-    executed_protocol_fees: &[Asset],
+    total_fee: Asset,
+    protocol_fees: &[Asset],
 ) -> Result<(), sqlx::Error> {
-    let (protocol_fee_tokens, protocol_fee_amounts): (Vec<_>, Vec<_>) = executed_protocol_fees
+    let (protocol_fee_tokens, protocol_fee_amounts): (Vec<_>, Vec<_>) = protocol_fees
         .iter()
         .map(|entry| (entry.token, entry.amount.clone()))
         .unzip();
 
     const QUERY: &str = r#"
-INSERT INTO order_execution (order_uid, auction_id, reward, surplus_fee, block_number, protocol_fee_tokens, protocol_fee_amounts)
-VALUES ($1, $2, $3, $4, $5, $6, $7)
+INSERT INTO order_execution (order_uid, auction_id, reward, total_fee, total_fee_token, block_number, protocol_fee_tokens, protocol_fee_amounts)
+VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
 ON CONFLICT (order_uid, auction_id)
-DO UPDATE SET reward = $3, surplus_fee = $4, block_number = $5, protocol_fee_tokens = $6, protocol_fee_amounts = $7
+DO UPDATE SET reward = $3, total_fee = $4, total_fee_token = $5, block_number = $6, protocol_fee_tokens = $7, protocol_fee_amounts = $8
 ;"#;
     sqlx::query(QUERY)
         .bind(order)
         .bind(auction)
         .bind(0.) // reward is deprecated but saved for historical analysis
-        .bind(Some(executed_fee))
+        .bind(Some(total_fee.amount))
```

Comment:
Why do we have an `Option` here? Do we still need that?

Reply:
I assume it's because when the field was created in the table it was marked as nullable, because it was NULL for market orders.
Now we always have the value.

Reply:
resolved